checker/src/main/java/org/checkerframework/checker/lock/qual/GuardedBy.java

```diff
@@ -38,10 +37,10 @@
 @SubtypeOf(GuardedByUnknown.class)
 @Documented
 @DefaultQualifierInHierarchy
-@DefaultFor({TypeUseLocation.EXCEPTION_PARAMETER, TypeUseLocation.UPPER_BOUND})
 @DefaultInUncheckedCodeFor({TypeUseLocation.PARAMETER})
-@ImplicitFor(
-        types = {
+@DefaultFor(
+        value = {TypeUseLocation.EXCEPTION_PARAMETER, TypeUseLocation.UPPER_BOUND},
+        typeKinds = {
```

Comment:
Same comment as for `NonNull.java` below: these `typeKinds` no longer serve a purpose, as this is already the default qualifier.

Reply:
Tests fail without these, I'm looking into it.

Reply:
These are required because the default for local variables is `@GuardedByUnknown`, but if the local variable is one of these type kinds, the default should be `@GuardedByUnknown`.

Reply:
Maybe add a comment? If one is used to CLIMB defaults, this is surprising.